src/libOpenImageIO/imageio.cpp

```diff
@@ -676,6 +688,20 @@ getattribute(string_view name, TypeDesc type, void* val)
         *(float*)val = IB_total_image_read_time;
         return true;
     }
+    if (name == "allowed_input_formats" && type == TypeString) {
+        if (allowed_input_format_list.empty()) {
+            return false;
+        }
+        *(ustring*)val = allowed_input_format_list;
+        return true;
+    }
+    if (name == "allowed_output_formats" && type == TypeString) {
+        if (allowed_output_format_list.empty()) {
+            return false;
+        }
```

Comment:
I still believe that we should not return false (which usually indicates that it's an invalid query -- like an attribute name that doesn't exist) for an attribute that exists but happens to be the empty string.